packages/rpc-transport/src/json-rpc-transport-types.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Public API.
+ */
+export type Transport<TApi> = TransportMethods<TApi>;
+export type ArmedTransport<Api, TResponse> = ArmedTransportMethods<Api, TResponse> & {
+    send(): Promise<TResponse>;
+};
+export type ArmedBatchTransport<Api, TResponses extends unknown[]> = ArmedBatchTransportMethods<Api, TResponses> & {
+    sendBatch(): Promise<TResponses>;
+};
+
+/**
+ * Private transport-building types.
+ */
+type TransportMethods<TApi> = {
+    [TMethodName in keyof TApi]: ArmedTransportReturner<TApi, ApiMethodImplementations<TApi, TMethodName>>;
+};
+type ArmedTransportMethods<TApi, TResponse> = {
+    [TMethodName in keyof TApi]: ArmedBatchTransportReturner<
+        TApi,
+        ApiMethodImplementations<TApi, TMethodName>,
+        [TResponse]
+    >;
+};
```

Comment:
Could this just be simplified to `type ArmedTransportMethods<TApi, TResponse> = ArmedBatchTransportMethods<TApi, [TResponse]>`?

Reply:
I think so! #1242.